src/jit/lclvars.cpp

```diff
@@ -1978,13 +1978,16 @@ Compiler::lvaStructFieldInfo Compiler::StructPromotionHelper::GetFieldInfo(CORIN
     {
         unsigned  simdSize;
         var_types simdBaseType = compiler->getBaseTypeAndSizeOfSIMDType(fieldInfo.fldTypeHnd, &simdSize);
-        if (simdBaseType != TYP_UNKNOWN)
+        if ((simdSize >= compiler->minSIMDStructBytes()) && (simdSize <= compiler->maxSIMDStructBytes()))
```

Comment:
This should have a comment indicating that we will only promote fields of SIMD types that fit into a SIMD register.